Applications/Spire/Source/Ui/ScrollArea.cpp

```diff
@@ -0,0 +1,165 @@
+#include "Spire/Ui/ScrollArea.hpp"
+#include <QHoverEvent>
+#include <QScrollBar>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Spire/Utility.hpp"
+
+using namespace Spire;
+
+namespace {
+  const auto MINIMUM_TABLE_WIDTH = 750;
+  const auto SCROLL_BAR_HIDE_TIME_MS = 500;
+  const auto SCROLL_BAR_MAX_SIZE = 13;
+  const auto SCROLL_BAR_MIN_SIZE = 6;
+}
+
+ScrollArea::ScrollArea(QWidget* parent)
+    : QScrollArea(parent) {
+  setMouseTracking(true);
+  horizontalScrollBar()->setContextMenuPolicy(Qt::NoContextMenu);
+  verticalScrollBar()->setContextMenuPolicy(Qt::NoContextMenu);
+  setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+  setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+  m_h_scroll_bar_timer.setInterval(SCROLL_BAR_HIDE_TIME_MS);
+  connect(&m_h_scroll_bar_timer, &QTimer::timeout, this,
+    &ScrollArea::hide_horizontal_scroll_bar);
+  m_v_scroll_bar_timer.setInterval(SCROLL_BAR_HIDE_TIME_MS);
+  connect(&m_v_scroll_bar_timer, &QTimer::timeout, this,
+    &ScrollArea::hide_vertical_scroll_bar);
+}
+
+void ScrollArea::setWidget(QWidget* widget) {
+  widget->setMouseTracking(true);
+  widget->setAttribute(Qt::WA_Hover);
+  widget->installEventFilter(this);
+  QScrollArea::setWidget(widget);
+}
+
+bool ScrollArea::eventFilter(QObject* watched, QEvent* event) {
+  if(event->type() == QEvent::HoverMove) {
+    auto e = static_cast<QHoverEvent*>(event);
+    if(is_within_horizontal_scroll_bar(e->pos().y()) &&
+        !verticalScrollBar()->isVisible()) {
+      set_scroll_bar_style(SCROLL_BAR_MAX_SIZE);
+      setHorizontalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+      setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    } else if(is_within_vertical_scroll_bar(e->pos().x()) &&
+        !horizontalScrollBar()->isSliderDown()) {
+      set_scroll_bar_style(SCROLL_BAR_MAX_SIZE);
+      setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOn);
+      setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    } else {
+      set_scroll_bar_style(SCROLL_BAR_MIN_SIZE);
+      if(!m_v_scroll_bar_timer.isActive() &&
```

Comment:
You can move `!m_v_scroll_bar_timer.isActive()` to an outer `if` check for both vertical and horizontal routines so that you do not have to check twice.